File: my-folder/0128-longest-consecutive-sequence/solution.cpp
class Solution {
public:
    int longestConsecutive(vector<int>& nums) {
        stack<int>st;

        if( nums.size() == 0) 
          return 0;
        int curr = 0, maxx = 0;

        sort(nums.begin(), nums.end());
        

        for(auto n: nums){
          if(!st.empty() && n-1 == st.top()){
            curr += 1;
            st.pop();
            st.push(n);
            maxx = max(maxx, curr);
          }
          else if(!st.empty() && st.top() == n) 
            continue;
          else {
            curr = 0;
            st.push(n);
          }
        }
        return maxx+1;
    }
};

File: my-folder/0128-longest-consecutive-sequence/solution.py
class Solution:
    def longestConsecutive(self, nums: List[int]) -> int:

        if not nums:
            return 0

        nums=list(set(nums))
        nums.sort()
        res=[]

        maxL = 0
        curL=0
        for n in nums:

            if res and res[-1] == n-1:
                curL += 1
                maxL=max(maxL,curL)
                res.append(n)
            else:

                curL=0
                res.append(n)
        return maxL+1




        


        






